artifacts.go

```diff
@@ -71,39 +70,51 @@ func read(aPath string) (parser.Workers, error) {
 	if ar == nil {
 		return nil, errors.New("Can not read artifact file.")
 	}
-	defer ar.Close()
 
 	p := parser.RootfsParser{
 		W: ioutil.Discard, // don't store update anywhere
 	}
 	ar.Register(&p)
 
-	return ar.Read()
+	_, err = ar.Read()
+	if err != nil {
+		return nil, err
+	}
+	defer ar.Close()
+
+	return ar, nil
```

Comment:
Not really. After reading the whole `ar` structure is filled in with the data. Later on we are using it simply. No further `Read()` is done.